creator-node/src/diskManager.js

```diff
@@ -332,6 +336,137 @@ class DiskManager {
     const redisSetKey = `${REDIS_DEL_FILE_KEY_PREFIX}${walletPublicKey}`
     await redisClient.del(redisSetKey)
   }
+
+  // lists all the folders in /file_storage/files/
+  static async listSubdirectoriesInFiles() {
+    const subdirectories = []
+    try {
+      const stdout = await this._execShellCommand(
+        `find ${path.join(this.getConfigStoragePath(), 'files')} -maxdepth 1`
+      )
+      // stdout is a string so split on newline and remove any empty strings
+      for (const dir of stdout.split('\n')) {
+        const dirTrimmed = dir.trim()
+        // if dirTrimmed is a non-null string and is not just equal to base directory
+        if (
+          dirTrimmed &&
+          dirTrimmed !== path.join(this.getConfigStoragePath(), 'files')
+        ) {
+          subdirectories.push(dirTrimmed)
+        }
+      }
+
+      return subdirectories
+    } catch (e) {
+      genericLogger.error(
+        `Error in diskManager - listSubdirectoriesInFiles: ${e}`
+      )
+    }
+  }
+
+  // list all the CIDs in /file_storage/files/AqN
+  static async listNestedCIDsInFilePath(filesSubdirectory) {
+    const cids = []
+    // find files older than 3 days in filesSubdirectory (eg /file_storage/files/AqN)
+    try {
+      const stdout = await this._execShellCommand(
+        `find ${filesSubdirectory} -mtime +${DAYS_BEFORE_PRUNING_ORPHANED_CONTENT}`
+      )
+
+      for (const file of stdout.split('\n')) {
+        const fileTrimmed = file.trim()
+        // if fileTrimmed is a non-null string and is not just equal to base directory
+        if (fileTrimmed && fileTrimmed !== filesSubdirectory) {
+          const parts = fileTrimmed.split('/')
+          cids.push(parts[parts.length - 1])
+        }
+      }
+
+      return cids
+    } catch (e) {
+      genericLogger.error(
+        `Error in diskManager - listNestedCIDsInFilePath: ${e}`
+      )
+    }
+  }
+
+  static async sweepSubdirectoriesInFiles() {
+    const subdirectories = await this.listSubdirectoriesInFiles()
+    if (!subdirectories) return
+
+    for (let i = 0; i < subdirectories.length; i += 1) {
+      try {
+        const subdirectory = subdirectories[i]
+        genericLogger.info(
+          `diskManager#sweepSubdirectoriesInFiles - iteration ${i} out of ${subdirectories.length}`
+        )
+        const cids = await this.listNestedCIDsInFilePath(subdirectory)
+
+        // TODO - parallelize into batches
```

Comment:
I see the PR is in draft so assuming the batch size and what-not will come later?